docs/config/ssr-options.md

```diff
@@ -2,17 +2,17 @@
 
 ## ssr.external
 
-- **Type:** `string[]`
+- **Type:** `string[] | true`
 - **Related:** [SSR Externals](/guide/ssr#ssr-externals)
 
-Force externalize dependencies for SSR.
+Externalize dependencies for SSR. By default, all dependencies are externalized, except linked dependencies for HMR. This can opted-out by adding the dependency to this option, or set `true` to force externalize all dependencies.
 
 ## ssr.noExternal
 
 - **Type:** `string | RegExp | (string | RegExp)[] | true`
 - **Related:** [SSR Externals](/guide/ssr#ssr-externals)
 
-Prevent listed dependencies from being externalized for SSR. If `true`, no dependencies are externalized.
+Prevent listed dependencies from being externalized for SSR. If `true`, no dependencies are externalized. If a dependency is explicitly defined in `ssr.external`, it will take priority and be externalized along with its transitive dependencies.
```

Comment:
we should clarify that this works both with packages names and with direct entries. also the part about ssr.external applying transitively should probably go in the section for ssr.external and the part about linked packages should probably go here, so I've swapped those in the suggestions
```suggestion
Bundles the given packages or modules within a package for SSR. If `true`, no dependencies are externalized. Linked dependencies will be bundled by default in order to support HMR on those packages.
```